### PR TITLE
3001 linux_distribution osrelease bugfix

### LINKED ISSUE #57781: distro/linux_distribution in 3001 breaks osrelease on Centos 7
*Description*

In grains/core.py, v.3001 importing linux_distribution has been changed from:
```

import warnings
....

try:
    from platform import linux_distribution as _deprecated_linux_distribution

    def linux_distribution(**kwargs):
        with warnings.catch_warnings():
            warnings.simplefilter("ignore")
            return _deprecated_linux_distribution(**kwargs)
   except ImportError:
       from distro import linux_distribution


```
to

```
from distro import linux_distribution
```

which results in os_release returning only the major release here (line 2035):

```
        (osname, osrelease, oscodename) = [
            x.strip('"').strip("'") for x in linux_distribution()
        ]

```
and failing to retrieve the correct osrelease and osrelease_info values:

```
    osrelease:
        7
    osrelease_info:
        - 7

```
This is occurring on Centos 7.4, salt v3001

salt versions:

Salt Version:
           Salt: 3001
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
       M2Crypto: 0.33.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.8 (default, Apr 21 2020, 17:13:27)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 17.0.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.2.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7 Core

### PR DESCRIPTION
**Pull request by marco-kusa**
### What does this PR do?
  
The bug that doesn't allow use to source the os version in Centos 7 from lsb_distrib_release affects Centos 8 as well.

The distro module was made a hard dependency in 3001 replacing platform, but cannot source the minor version in Centos 7/8 via the linux_distribution function, as this calls version()
    without best=True. 

We switch to calling the functions ourselves and using best=True to get distro to use /etc/centos-release to retrieve the full minor version.

Comments have been updated to reflect this and the fact that the platform module is not used anymore.

### What issues does this PR fix or reference?
Fixes #57781

### Previous Behavior
We import linux_distribution from distro but on Centos 7/8 osrelease contains only the major version.
We do not use the information sourced from distro to get the correct minor version on Centos 8.

### New Behavior
We import distro and rewrite linux_distribution with our own calls to distro.id(), distro.version(best=True), distro.codename()
where distro.version(best=True) now returns the correct minor version information. 
We use this behavior for Centos 8 as well.

### CODE REVIEW

**Comment by mcuca1** [comment]
I can see there are some form checks in the pre-commit I am not passing, are these changes mandatory?

**Review by Akm0d — DISMISSED** [review]

**Review by cmcmarrow — DISMISSED** [review]

**Review by garethgreenaway — DISMISSED** [review]

**Comment by cmcmarrow** [comment]
@mcuca1 thanks for the PR and tests. Sorry for the later review.

**Review by krionbsd — DISMISSED** [review]

File: salt/grains/core.py [hunk 1]
```diff
@@ -40,11 +41,20 @@
...
 import salt.utils.platform
 import salt.utils.stringutils
-from distro import linux_distribution
 from salt.ext import six
 from salt.ext.six.moves import range
 from salt.utils.network import _get_interfaces
 
+
+# rewrite distro.linux_distribution to allow best=True kwarg in version(), needed to get the minor version numbers in CentOS
+def linux_distribution():
```
Comment (s0undt3ch) [inline salt/grains/core.py:50]: Please prefix this function with `_` since it should not be a _public_ function.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/grains/core.py (hunk 1 above)
Comment (mcuca1) [reply salt/grains/core.py]: done

**Review by s0undt3ch — DISMISSED** [review]

**Comment by s0undt3ch** [comment]
That was fast! Thanks!

**Comment by mcuca1** [comment]
> 
> 
> That was fast! Thanks!

thank you!

**Review by dwoz — APPROVED** [review]

**Pull request merged**